api/tests/test_api_spec.py

```diff
@@ -0,0 +1,156 @@
+from conftest import AUTH_COOKIE, DEFAULT_USERNAME, JOBS_URI, auth, login, make_job, submit_batch
+from flask_api import status
+
+
+def test_not_logged_in(client):
+    response = submit_batch(client)
```

Comment:
think this can be `response = client.post(JOBS_URI)`, actually, it should check auth before checking the payload

Reply:
`test_not_logged_in()` and `test_jobs_bad_method()` could both reduce to either of these, I kind of like the `open()` approach:

```
    for method in [client.get, client.post, client.head]:
        response = method(JOBS_URI)
        assert response.status_code == status.HTTP_401_UNAUTHORIZED

    for method in ['POST', 'GET', 'HEAD']:
        response = client.open(JOBS_URI, method=method)
        assert response.status_code == status.HTTP_401_UNAUTHORIZED
```